sungc/rasterio_funcs.py

```diff
@@ -1,38 +1,55 @@
 #!/usr/bin/env python3
 
-import os
 import fiona
+import warnings
 import rasterio
 import rasterio.mask
 import numpy as np
 
-from rasterio.warp import reproject
+from typing import Union
+from pathlib import Path
+from rasterio import DatasetReader
+from rasterio.warp import reproject, Resampling
 
 
-def get_basename(filename):
+def check_image_singleval(image: np.ndarray, value: Union[float, int], img_name: str):
+    """
+    Checks if numpy image has only a single value. If so, raises Exception
+    """
+    if np.isnan(value):
+        if not ~np.isnan(image).any():
+            # if there are no non-nan pixels (if all pixels are np.nan)
+            raise Exception(f"{img_name} only contains a single value ({value})")
+
+    else:
+        if np.all(image == value):
+            raise Exception(f"{img_name} only contains a single value ({value})")
+
+
+def get_basename(filename: Path):
     """ Get a file's basename without extension """
-    return os.path.basename(os.path.splitext(filename)[0])
+    return filename.stem
```

Comment:
A superfluos function now as the input is a `Path` object, and where only calling a property on the `Path` object, namely `.stem`

Reply:
This function was removed